### DISCUSSION TITLE
Generating hooks for overloaded functions (two fn with same name, but different args)

### DISCUSSION THREAD

**Discussion by hardingjam** [wevm/wagmi#3301] [Q&A]
I have two functions in my abi called "mint", one that takes 2 args, and one that takes 4. I used the CLI to generate a set of contract hooks, but it only generated one function called usePrepareMyContractMint, and it accepts two args. The ABI shows both functions from the contract.

Is there a way to generate hooks for the overloaded function? How can. specify which version of the version I want to use, when calling usePrepareContractWrite from @wagmi/react?

Or, to be more succinct - how does wagmi handle overload function calls such as this example from ethers.js :

```
         const tx = await contract
                            .connect(signer)
                            ["mint(uint256,address,uint256,bytes)"](shares, account, number, data); 
```

                           
When another mint function exists that accepts two args?

**Comment by hardingjam** [comment]
https://github.com/wevm/wagmi/issues/2724
https://github.com/wevm/wagmi/issues/2297

A couple of issues mention having functions with similar names, but I can't see anything about overloads. This answer seems to no longer be accurate when using @wagmi/react:

https://ethereum.stackexchange.com/questions/133827/how-to-use-overloaded-function-from-smart-contract-with-wagmi

**Comment by tmm** [comment]
You should be able to just call the function with the name as normal and internally Wagmi (via Viem) will identify which overload is called.
